index.html

```diff
@@ -2326,6 +2327,102 @@ <h2>Definition of Roles</h2>
 				</tbody>
 			</table>
 		</div>
+		<div class="role" id="comment">
+			<rdef>comment</rdef>
+			<div class="role-description">
+				<p>A comment contains a remark expressing a reaction to other content within the page.</p>
+				<p>The relationship of comments to other content SHOULD be exposed as follows:</p>
+				<ol>
+					<li>If the comment refers to other content within the page, the source content SHOULD use <pref>aria-details</pref> to point to the related comment.</li>
+					<li>If the comment is a reply to another comment:
+						<ul>
+							<li>If all ancestor comments are available in the DOM, the reply comment SHOULD be a semantic descendant of the comment it relates to, either as a DOM descendant element, or via <pref>aria-owns</pref>. In this case, user agents MUST automatically compute the level and expose it to assistive technologies. This is the preferred method for defining
+								the comment hierarchy, as it creates a single root comment that can be referred to by commented content via <pref>aria-details</pref>.</li>
+							<li>Alternatively, if all ancestor comments are not in the DOM, such as when comments are paginated, the hierarchical
```

Comment:
This alternative is unclear because it implies that aria-owns or DOM hierarchy is not used to relate the comments and replies that are present in the DOM. So, if DOM hierarchy is not being used in this alternative, and if there is a reply to a reply present in the DOM, then how are those replies related to one another?

Separately, this should mention aria-posinset and aria-setsize. I've added that. This suggestion, however, does not address the structural ambiguity.

```suggestion
							<li>Alternatively, if all ancestor comments are not in the DOM, such as when comments are paginated, specify the hierarchical level with <pref>aria-level</pref>, the position of the comment with in that level with <pref>aria-posinset</pref>, and the number of comments at the same level with <pref>aria-setsize</pref>.</li>
```

Reply:
It's like tree view. In this case, the structure, posinset and setsize are all implied by the aria-level. The author can use aria-posinset and aria-setsize, but they don't have to. If they only use aria-level, the posinset and setsize can be computed by the user agent. It's clear when the set changes because the level goes up. This works in Chrome and Firefox for tree items, and in Chrome for comment.

Example:
Level 1  (posinset = 1, setsize = 1) 
Level 2 (posinset = 1, setsize = 2)
Level 3 (posinset = 1, setsize = 1)
Level 2 (posinset = 2, setsize = 2)

Reply:
From the user agent perspective, tje only reason to point to a container element of comments is for the aria-level case. You can use as many ids in aria-details as you want.

It might be more convenient to point to a containing element though, because if comments are being constantly added and removed, it's no longer necessary to update aria-details. That's a separate issue though, from what is being commented (e.g. a blog article). I think this new text confuses all of that. I'll try to fix.